### PR TITLE
Add flag to disable propagation of user-defined labels and annotations

### PR DESCRIPTION
**Pull request by thedadams**
The user is still free to attempt to add labels and annotations to the various objects, but the controller will ignore them when creating the other Kubernetes objects. If the flag is later turned off (so labels and annotations are allowed to propagate) then new objects will be created with the desired labels and annotations.

Two tests are also added to test the labels and annotations functionality with and without propagation.

The flag added is sort of long, but I wanted to ensure that it was explicit about what it was doing.

Issue: https://github.com/acorn-io/acorn/issues/1041

### CODE REVIEW

File: pkg/controller/appdefinition/deploy.go [hunk 1]
```diff
@@ -32,6 +32,22 @@ import (
...
 
+func RemoveLabelsAndAnnotationsConfig(h router.Handler) router.Handler {
+	return router.HandlerFunc(func(req router.Request, resp router.Response) error {
+		appInstance := req.Object.(*v1.AppInstance)
+		cfg, err := config.Get(req.Ctx, req.Client)
+		if err != nil {
+			return err
+		}
+
+		if cfg.IgnoreUserLabelsAndAnnotations != nil && *cfg.IgnoreUserLabelsAndAnnotations {
```
Comment (cjellick) [inline pkg/controller/appdefinition/deploy.go:43]: A tip @ibuildthecloud gave me: For nil bools feel free to add GetIgnoreUserLabelsAndAnnotations() method like the GetDevMode() method on appInstanceSpec. Makes the code more readable.

File: pkg/controller/appdefinition/testdata/deployspec/labels/existing.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
```
Comment (cjellick) [inline pkg/controller/appdefinition/testdata/deployspec/labels/existing.yaml:1]: Why did you need to add this file?

Comment (cjellick) [reply pkg/controller/appdefinition/testdata/deployspec/labels/existing.yaml:1]: wait, this whole dir is new. its a whole new test?

Comment (cjellick) [reply pkg/controller/appdefinition/testdata/deployspec/labels/existing.yaml:1]: resolving, disregard

**Review by cjellick — CHANGES_REQUESTED** [review]
Just one small nitpck

File: pkg/controller/appdefinition/deploy.go (hunk 1 above)
Comment (thedadams) [reply pkg/controller/appdefinition/deploy.go]: I took a slightly different approach here because `config.Get` "completes" the config object with defaults.

**Review by jsilverio22 — APPROVED** [review]

File: pkg/controller/appdefinition/deploy.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/controller/appdefinition/deploy.go]: for my benefit..this is just to check for bools in AppInstanceSpec are there and set?

Comment (cjellick) [reply pkg/controller/appdefinition/deploy.go]: Which part @jsilverio22 ? Really anywhere we have nilable bools, you can add a helper to do this type of logic, EXCEPT for config because as donnie is pointing out, it isnt necessary because the config.Get is responsible for ensure its not nil

**Review by cjellick — APPROVED** [review]

File: pkg/controller/appdefinition/deploy.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/controller/appdefinition/deploy.go]: I was looking at the GetDevMode(). But that makes sense bundling it up for clarity and why it is not necessary in this case.
